src/server/ua_subscription_events.c

```diff
@@ -373,25 +391,46 @@ UA_Event_addEventToMonitoredItem(UA_Server *server, const UA_NodeId *event,
 }
 
 static const UA_NodeId objectsFolderId = {0, UA_NODEIDTYPE_NUMERIC, {UA_NS0ID_OBJECTSFOLDER}};
+#ifdef UA_ENABLE_SUBSCRIPTIONS_ALARMS_CONDITIONS
+static const UA_NodeId parentReferences_events[4] =
+#else
 static const UA_NodeId parentReferences_events[2] =
+#endif /*UA_ENABLE_SUBSCRIPTIONS_ALARMS_CONDITIONS*/
     {{0, UA_NODEIDTYPE_NUMERIC, {UA_NS0ID_ORGANIZES}},
-     {0, UA_NODEIDTYPE_NUMERIC, {UA_NS0ID_HASCOMPONENT}}};
-
+     {0, UA_NODEIDTYPE_NUMERIC, {UA_NS0ID_HASCOMPONENT}},
+#ifdef UA_ENABLE_SUBSCRIPTIONS_ALARMS_CONDITIONS
+     /* check also other hierarchical references */
+     {0, UA_NODEIDTYPE_NUMERIC, {UA_NS0ID_HASEVENTSOURCE}},
+     {0, UA_NODEIDTYPE_NUMERIC, {UA_NS0ID_HASNOTIFIER}}
+#endif /*UA_ENABLE_SUBSCRIPTIONS_ALARMS_CONDITIONS*/
+    };
 UA_StatusCode
 UA_Server_triggerEvent(UA_Server *server, const UA_NodeId eventNodeId, const UA_NodeId origin,
                        UA_ByteString *outEventId, const UA_Boolean deleteEventNode) {
-    /* Check that the origin node exists */
-    const UA_Node *originNode = UA_Nodestore_getNode(server->nsCtx, &origin);
-    if(!originNode) {
-        UA_LOG_ERROR(&server->config.logger, UA_LOGCATEGORY_USERLAND,
-                     "Origin node for event does not exist.");
-        return UA_STATUSCODE_BADNOTFOUND;
+#ifdef UA_ENABLE_SUBSCRIPTIONS_ALARMS_CONDITIONS
+    UA_Boolean isCallerAC = false;
+    if(isConditionOrBranch(server, &eventNodeId, &origin, &isCallerAC)) {
```

Comment:
`isConditionOrBranch` has three nested `LIST_FOREACH`, so if someone enables A&C this will traverse condition lists (that might be quite long with branches) every time they want to trigger an event. And `isCallerAC` field is updated throughout the rest of the code.
As I understand it, all that is needed just to make sure that a user won't trigger a condition event using this function. As far as I am concerned this seems like overkill.

Reply:
@andranik-s thank you for reviewing the code.
I agree with you on this point, that it isn't a nice solution. I discussed that already with @jpfr in:
https://github.com/open62541/open62541/pull/2083#discussion_r248270489

The Problem is when a user triggers a Condition Event using the normal `UA_Server_triggerEvent` function, we will lose track of the last EventId and we won't be able to use the Condition Methods.
another solution would be to create an extra flat list to store all the Condition and their Branches NodeId's, but the pitfall would be that we hav to manage two lists...
Maybe you have a better Idea?:)

Reply:
@sqadasi IMHO this check is not nesessary at all. Just warning users in documentation that this function should not be used with condition events seems OK to me.

Alternatively, event type could be used to check for conditions. It will require to create an additional (static) function to trigger condition events inside your code, but I don't think it is a problem. Though, I am not sure it is a very good idea because it will not let users trigger "raw" condition events should they want that for whatever reason.

Reply:
@andranik-s thank you for answering and sorry for my late respond...
By using the EventType we have to do two steps. First getting the EventType NodeId and then checking the path of it, whether it's a SubType of ConditionType. Here we could get the problem of iterating through the Node path with `getTypesHierarchy`, which will be used for the second time in the Event Triggering function and I think it won't be more effective if we have deeply nested ConditionTypes.

IMHO I think there will be always someone who will use the wrong function by mistake, even with a good documented API, so I am not sure whether it's a good idea to remove that check at all :)

Maybe @jpfr and @Pro could share their opinions with us:)